app/src/main/java/com/example/news/ui/home/HomeRVAdapter.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.news.ui.home
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.example.Articles
+import com.example.news.R
+import de.hdodenhof.circleimageview.CircleImageView
+
+class HomeRVAdapter(var context: Context,var articles: ArrayList<Articles>,var onHomeRvClickListener: OnHomeRvClickListener) : RecyclerView.Adapter<HomeRVAdapter.Holder>() {
```

Comment:
you can do the `onHomeRvClickListener` in a cleaner way something like
```
class HomeRVAdapter(var context: Context,var articles: ArrayList<Articles>,var onHomeRvClickListener: (Articles)-> Unit)
```
and delete the interface?